src/nvim/drawline.c

```diff
@@ -286,10 +286,12 @@ static void draw_virt_text(win_T *wp, buf_T *buf, int col_off, int *end_col, int
       int vcol = item->draw_col - col_off;
       col = draw_virt_text_item(buf, item->draw_col, vt->data.virt_text,
                                 vt->hl_mode, max_col, vcol);
-    }
-    item->draw_col = INT_MIN;  // deactivate
-    if (vt && vt->pos == kVPosEndOfLine && do_eol) {
-      state->eol_col = col + 1;
+      if (!(vt->flags & kVTRepeatLinebreak) || item->kind == kDecorKindUIWatched) {
```

Comment:
The `item->kind == kDecorKindUIWatched` condition is always false here because `vt` is always `NULL` is that case.